src/global-state/dealsFeed.tsx

```diff
@@ -126,25 +138,41 @@ function dealsFeedReducer(
   }
 }
 
-async function refreshDealsFeed(
+async function refreshNewFeed(
   fetchFeed: FeedFetcher,
   dispatch: Dispatch,
-  dealsFeed?: DealsFeed,
 ): Promise<void> {
   dispatch({ type: 'refresh' });
 
   const newFeed = await fetchFeed();
 
   dispatch({
     type: 'set',
-    dealsFeed: dealsFeed
-      ? new DealsFeed(dealsFeed.feed, newFeed)
-      : new DealsFeed(newFeed),
+    dealsFeed: new DealsFeed(newFeed),
+  });
+}
+
+async function loadFeedNextPage(
+  fetchFeed: FeedFetcher,
+  dispatch: Dispatch,
+  dealsFeed: DealsFeed,
+): Promise<void> {
+  dispatch({ type: 'refresh' });
+
+  const { lastFetchedPage } = dealsFeed;
+  const newFeed = await fetchFeed(lastFetchedPage + 1);
+
+  dispatch({
+    type: 'set',
+    dealsFeed: new DealsFeed(dealsFeed.feed, newFeed, lastFetchedPage + 1),
```

Comment:
Would be better to attach the `fetchFeed` method used to create the `DealsFeed` to the `DealsFeed` for it to use internally to fetch the next page, rather than mmanaging next-page counting externally.